cli/cppcheckexecutor.cpp

```diff
@@ -339,27 +337,39 @@ bool CppCheckExecutor::reportUnmatchedSuppressions(const Settings &settings, con
     assert(!(!files.empty() && !fileSettings.empty()));
 
     // bail out if there is a suppression of unmatchedSuppression which matches any file
-    const auto suppr = suppressions.getSuppressions();
+    auto suppr = suppressions.getSuppressions();
     if (std::any_of(suppr.cbegin(), suppr.cend(), [](const SuppressionList::Suppression& s) {
         return s.errorId == "unmatchedSuppression" && (s.fileName.empty() || s.fileName == "*") && s.lineNumber == SuppressionList::Suppression::NO_LINE;
     }))
         return false;
 
+    SuppressionList supprlist;
+
+    const bool doUnusedFunctionOnly = Settings::doUnusedFunctionOnly();
+    // ignore all other suppressions if we use the unusedFunction hack
+    for (auto&& s : suppr)
```

Comment:
I guess some C++ algorithm can be used for instance `std::copy_if` or `std::erase_if`.